dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/hooks/PreCheckDataRelationsValidationHookTest.java

```diff
@@ -60,37 +64,323 @@
 import org.mockito.junit.MockitoJUnit;
 import org.mockito.junit.MockitoRule;
 
+import com.google.api.client.util.Maps;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+
 /**
  * @author Enrico Colasante
  */
-public class PreCheckDataRelationsValidationHookTest
+public class PreCheckDataRelationsValidationHookTest extends DhisConvenienceTest
 {
+    private static final String PROGRAM_WITHOUT_REGISTRATION_ID = "PROGRAM_WITHOUT_REGISTRATION_ID";
+
+    private static final String PROGRAM_WITH_REGISTRATION_ID = "PROGRAM_WITH_REGISTRATION_ID";
+
+    private static final String PROGRAM_STAGE_ID = "PROGRAM_STAGE_ID";
+
+    private static final String ORG_UNIT_ID = "ORG_UNIT_ID";
+
+    private static final String ANOTHER_ORG_UNIT_ID = "ANOTHER_ORG_UNIT_ID";
+
+    private static final String TEI_TYPE_ID = "TEI_TYPE_ID";
+
+    private static final String TEI_ID = "TEI_ID";
+
+    private static final String ANOTHER_TEI_TYPE_ID = "ANOTHER_TEI_TYPE_ID";
+
+    private static final String ANOTHER_TEI_ID = "ANOTHER_TEI_ID";
+
+    private static final String ENROLLMENT_ID = "ENROLLMENT_ID";
+
+    private static final String ANOTHER_ENROLLMENT_ID = "ANOTHER_ENROLLMENT_ID";
+
     private PreCheckDataRelationsValidationHook validatorToTest;
 
     @Rule
     public MockitoRule mockitoRule = MockitoJUnit.rule();
 
-    @Mock
-    private ProgramInstanceService programInstanceService;
-
     @Mock
     private CategoryService categoryService;
 
     @Mock
     private TrackerImportValidationContext ctx;
 
-    @Mock
     private TrackerBundle bundle;
 
+    @Mock
+    private TrackerPreheat preheat;
+
     private ValidationErrorReporter reporter;
 
     @Before
     public void setUp()
     {
         validatorToTest = new PreCheckDataRelationsValidationHook( categoryService );
+        bundle = TrackerBundle.builder().preheat( preheat ).build();
 
         when( ctx.getBundle() ).thenReturn( bundle );
-        when( bundle.getValidationMode() ).thenReturn( ValidationMode.FULL );
+
+        OrganisationUnit organisationUnit = createOrganisationUnit( 'A' );
+        organisationUnit.setUid( ORG_UNIT_ID );
+        when( ctx.getOrganisationUnit( ORG_UNIT_ID ) ).thenReturn( organisationUnit );
+
+        OrganisationUnit anotherOrganisationUnit = createOrganisationUnit( 'B' );
+        anotherOrganisationUnit.setUid( ANOTHER_ORG_UNIT_ID );
+        when( ctx.getOrganisationUnit( ANOTHER_ORG_UNIT_ID ) ).thenReturn( anotherOrganisationUnit );
+
+        TrackedEntityType trackedEntityType = createTrackedEntityType( 'A' );
+        trackedEntityType.setUid( TEI_TYPE_ID );
+        when( ctx.getTrackedEntityType( TEI_TYPE_ID ) ).thenReturn( trackedEntityType );
+
+        TrackedEntityType anotherTrackedEntityType = createTrackedEntityType( 'B' );
```

Comment:
Set up class can be very long to read, i would suggest  to add relevant chunck of info in the test method itself even if we take the risk of being redundant

Reply:
Here the setup is big because we are validating quite different things.
Proper solution would be to divide the validators.
I reorganised the setup to be more modular